spec/unit/Java.spec.js

```diff
@@ -0,0 +1,158 @@
+            process.env.JAVA_HOME = '/tmp/jdk';
+
+            await Java._ensure();
+
+            expect(process.env.PATH.split(path.delimiter))
+                .toContain(path.join(process.env.JAVA_HOME, 'bin'));
+        });
+
+        it('without JAVA_HOME / with javac - Mac OS X - success', async () => {
+            spyOn(utils, 'forgivingWhichSync').and.returnValue('/tmp/jdk/bin');
+            process.env.JAVA_HOME = '';
+            const fsSpy = jasmine.createSpy('fs').and.returnValue(true);
+            Java.__set__('fs', {
+                existsSync: fsSpy
+            });
+            Java.__set__('execa', jasmine.createSpy('execa').and.returnValue(Promise.resolve({
+                stdout: '/tmp/jdk'
+            })));
+
+            await Java._ensure();
+
+            expect(fsSpy).toHaveBeenCalledWith('/usr/libexec/java_home');
+            expect(process.env.JAVA_HOME).toBe('/tmp/jdk');
+        });
+
+        it('without JAVA_HOME / with javac - Mac OS X - error', async () => {
+            spyOn(utils, 'forgivingWhichSync').and.returnValue('/tmp/jdk/bin');
+            process.env.JAVA_HOME = '';
+            Java.__set__('fs', {
+                existsSync: jasmine.createSpy('fs').and.returnValue(true)
+            });
+            Java.__set__('execa', jasmine.createSpy('execa').and.returnValue(Promise.reject({
+                shortMessage: 'test error'
+            })));
+            const emitSpy = jasmine.createSpy('events.emit');
+            Java.__set__('events', {
+                emit: emitSpy
+            });
+
+            await expectAsync(Java._ensure()).toBeRejectedWithError(CordovaError, /Failed to find 'JAVA_HOME' environment variable/);
+
+            expect(emitSpy).toHaveBeenCalledWith('verbose', 'test error');
+        });
+
+        it('derive from javac location - success', async () => {
+            spyOn(utils, 'forgivingWhichSync').and.returnValue('/tmp/jdk/bin');
+            process.env.JAVA_HOME = '';
+            Java.__set__('fs', {
+                existsSync: jasmine.createSpy('fs').and.callFake((path) => {
+                    if (/java_home$/.test(path)) {
+                        return false;
+                    } else {
+                        return true;
+                    }
+                })
+            });
+
+            await Java._ensure();
+
+            expect(process.env.JAVA_HOME).toBe('/tmp');
+        });
+
+        it('derive from javac location - error', async () => {
+            spyOn(utils, 'forgivingWhichSync').and.returnValue('/tmp/jdk/bin');
+            process.env.JAVA_HOME = '';
+            Java.__set__('fs', {
+                existsSync: jasmine.createSpy('fs').and.callFake((path) => {
+                    if (/java_home$/.test(path)) {
+                        return false;
+                    } else {
+                        return false;
+                    }
+                })
+            });
```

Comment:
AFAICT this should be the same fake implementation as above
```suggestion
            Java.__set__('fs', { existsSync: path => !/java_home$/.test(path) });
```